openmdao/utils/assert_utils.py

```diff
@@ -0,0 +1,108 @@
+            For 'J_fd', 'J_fwd', 'J_rev' the value is: A numpy array representing the computed
+                Jacobian for the three different methods of computation.
+    atol : float
+        absolute error. Default is 1e-6.
+    rtol : float
+        relative error. Default is 1e-6.
+    """
+    desired = (0.0, 0.0, 0.0)
+    for comp in data:
+        for (var, wrt) in data[comp]:
+            for error_type, tolerance in [('abs error', atol), ('rel error', rtol), ]:
+                actual = data[comp][var, wrt][error_type]
+                if not np.isnan(actual).any():
+                    assert_allclose(actual, desired, atol=tolerance,
+                                    err_msg='{0} error in partial of'
+                                            ' {1} wrt {2} in component {3}'.format(error_type,
+                                                                                   var, wrt, comp),
+                                    verbose=True)
+
+
+def assert_no_approx_partials(system, include_self=True, recurse=True):
+    """
+    Raise assertion error if any component within system is using approximated partials.
+
+    Parameters
+    ----------
+    system : System
+        The system under which to search for approximated partials.
+    include_self : bool
+        If True, include this system in the iteration.
+    recurse : bool
+        If True, iterate over the whole tree under this system.
+
+    Raises
+    ------
+    AssertionError
+        If a subsystem of group is found to be using approximated partials.
+
+    """
+    approximated_partials = {}
+    msg = 'The following components use approximated partials:\n'
+    for s in system.system_iter(include_self=include_self, recurse=recurse):
+        if isinstance(s, Component):
+            if s._approximated_partials:
+                approximated_partials[s.pathname] = s._approximated_partials
+                msg += '    ' + s.pathname + '\n'
+                for partial in s._approximated_partials:
+                    msg += '        of={0:12s}    wrt={1:12s}    method={2:2s}\n'.format(*partial)
+    if approximated_partials:
+        raise AssertionError(msg)
+
+
+def assert_no_dict_jacobians(system, include_self=True, recurse=True):
+    """
+    Raise an assertion error if any Group within system is found to be using dictionary jacobians.
+
+    Parameters
+    ----------
+    system : System
+        The system under which to search for approximated partials.
+    include_self : bool
+        If True, include this system in the iteration.
+    recurse : bool
+        If True, iterate over the whole tree under this system.
+
+    Raises
+    ------
+    AssertionError
+        If a subsystem of group is found to be using approximated partials.
+
+    """
+    groups_with_dict_jacobians = []
+    msg = 'The following groups use dictionary jacobians:\n'
+    for s in system.system_iter(include_self=include_self, recurse=recurse):
```

Comment:
system_iter has a 'typ' arg that will cause it to yield only systems that are instances of the specified type, so you could just put typ=Group in there and get rid of the if isinstance(s, Group) block.

Reply:
thanks for the tip. Done